mailbox/plugin/quota-mailing/src/main/java/org/apache/james/mailbox/quota/model/QuotaThresholdChanges.java

```diff
@@ -0,0 +1,99 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.quota.model;
+
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import org.apache.james.mailbox.quota.CompareWithCurrentThreshold;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterables;
+
+public class QuotaThresholdChanges {
+
+    private final ImmutableList<QuotaThresholdChange> changes;
+
+    public QuotaThresholdChanges() {
+        this(ImmutableList.of());
+    }
+
+    public QuotaThresholdChanges(QuotaThresholdChange... changes) {
+        this.changes = ImmutableList.copyOf(Arrays.asList(changes));
+    }
+
+    public QuotaThresholdChanges(List<QuotaThresholdChange> changes) {
+        this.changes = ImmutableList.copyOf(changes);
+    }
+
+    public CompareWithCurrentThreshold compareWithCurrentThreshold(QuotaThreshold currentThreshold, Duration gracePeriod) {
```

Comment:
date handling should be done in another step or maybe the change could contain the computed Duration before the state changed ?

Reply:
Not really... We need to see if that threshold appeared in recent history. Not just the date since last change.

Overwhise you might loose email if an account is feeling fast. 

IE:

```
 + h0 Threshold _75
 + h2 Threshold _90
 + h4 Threshold _98
```

In that case, even if all these changes were all made within the grace period, we still need to email them.